contracts/test/Multisig.test.ts

```diff
@@ -348,14 +348,9 @@ describe("Multisig with single owner", function () {
         tx_index: txIndex,
       });
 
-      try {
-        await nonOwner.invoke(multisig, "execute_transaction", {
-          tx_index: txIndex,
-        });
-        expect.fail("Should have failed");
-      } catch (err: any) {
-        assertErrorMsg(err.message, "not owner");
-      }
+      await nonOwner.invoke(multisig, "execute_transaction", {
```

Comment:
shouldn't this revert?

Reply:
Making this part of the review.

Reply:
This is an interesting approach. I assume you had an internal discussion about this - is it somewhere visible (in your own PRs for example)?
I see three options:
1) Keep it as it was before this PR
2) Change as you are suggesting
3) Make the transaction auto-execute upon last required confirmation.

If your idea is to make the process a bit easier (anyone can execute, so that could be performed by some untrusted backend of yours for example), then I would suggest option 3 instead. But do let me know about the reasoning behind this.

Reply:
The use case is for when one has a different set of signers (governance role) vs. executors (operations role). Yes, it also enables automation via protocols like Chainlink Keepers or custom infrastructure.

Option 2 vs. 3, we prefer option 2, as option 3 would require the last signer to pay gas fees for the transaction, which could be an order of magnitude larger than just voting on the proposal.

- It would introduce a game where signers would rush, then wait to avoid being last (probably a bad situation overall)
- It would make accounting more difficult, as there are use cases where operationally one doesn't want to pay for execution gas from the signer account but from another set of accounts.

This behavior is available from Serum multisig on Solana, and I also think Gnosis supports this workflow.

Reply:
Gnosis has this concept of "designated signer", who can propose transactions but not sign them - unsure of execution. But anyway, I now understand this better and it sounds fine. Thanks!